leash.go

```diff
@@ -525,3 +528,74 @@ func logStats(stats *responseStats, interval uint) {
 		stats.logAndReset()
 	}
 }
+
+// Final helper for outputting helpful log lines at the end of honeytail
+// execution. Calls private `responseStats` members indiscriminately
+// because we know this is only called at the end, after all goroutines
+// have exited.
+func outputHelp(options GlobalOptions, stats *responseStats, teamSlug string) {
+	// Build up Honeycomb URL for nicer logs
+	u, _ := url.Parse("https://ui.honeycomb.io")
+	u.Path = path.Join(u.Path, teamSlug, "datasets", options.Reqs.Dataset)
+
+	successes := 0
+	for code, count := range stats.totalStatusCodes {
+		if code >= 200 && code < 300 {
+			successes += count
+		}
+	}
+
+	// Check the average lag for the last interval. If it's over the
+	// AvgLagThreshold, warn the user.
+	if !options.Backfill && stats.cumulAvgLag > reporting.AvgLagThreshold {
+		logrus.Warn("Psst -- looks like you transmitted a lot of events with old timestamps!")
+		logrus.Warn("  (The average was around ", stats.cumulAvgLag, ")")
+		logrus.Warn("If you didn't mean to send events timestamped over ", reporting.AvgLagThreshold, " ago, this process may be failing to keep up with the input stream.\n")
+	}
+
+	// Nothing bad happened, yay
+	if successes > 0 {
```

Comment:
This decision tree of what to print on exit needs a flow diagram or something. All the nested if/else/elsif/else is making me get lost.  Would it be easier as a switch/case statement or do we just suffer and accept? Or something different?

I'm ok with it as is, just wondering if there's a not-terrible way to pivot to something else.